### ISSUE TITLE
Wrong link order warning does not de-duplicate

### ISSUE THREAD

**Issue by johnwhitington** [ocaml/ocaml#12527]
For example:

```
File "_none_", line 1:
Error: Wrong link order: Cpdfdrawcontrol depends on Cpdfshape,
         Cpdfdrawcontrol depends on Cpdfcolours,
         Cpdfdrawcontrol depends on Cpdfshape,
         Cpdfdrawcontrol depends on Cpdfcolours
```

This is in 4.14.1.

(In addition, it would be nicer to reformat this message so the errors line up.)

**Comment by hhugo** [comment]
X-ref #12084

**Comment by Octachron** [comment]
This should be fixed by #12084 in OCaml 5.3 .
